webapp/tests/test_functions.py

```diff
@@ -3864,3 +3862,138 @@ def test_summarize_1minute_alignToFrom(self):
               },
               seriesList, "1minute", func, True)
           self.assertEqual(result, expectedResults[func])
+
+    def test_exponentialMovingAverage_integerWindowSize(self):
```

Comment:
@iksaif I did some work on your second request yesterday, I would like to get your feedback to make sure it is in-line with what you are requesting before I refactor everything, and commit. 

Thanks,
Chris

- extracted gen_seriesList to global helper function
- created global function to create request contexts

```python
    def test_exponentialMovingAverage_evaluateTokens_returns_half_none(self):
        seriesList = self._gen_series_list_with_data(start=10)
        expectedResults = self._gen_series_list_with_data(
            key='exponentialMovingAverage(collectd.test-db0.load.value,10)',
            start=20,
            end=30,
            data=[0, 0.0, 0.182, 0.512, 0.965, 1.517, 2.15, 2.85, 3.604, 4.404, 5.239]
        )

        def mock_evaluateTokens(reqCtx, tokens, replacements=None):
            return self._gen_series_list_with_data(key='collectd.test-db0.load.value',start=10, end=30, data=([None] * 10 + range(0, 10)))

        with patch('graphite.render.functions.evaluateTokens', mock_evaluateTokens):
            result = functions.exponentialMovingAverage(self._build_request_context(endTime=datetime(1970, 1, 1, 0, 9, 0, 0, pytz.timezone(settings.TIME_ZONE))), seriesList, 10)

        self.assertEqual(result, expectedResults)

    def _build_request_context(self, startTime=datetime(1970, 1, 1, 0, 0, 0, 0, pytz.timezone(settings.TIME_ZONE)), endTime=datetime(1970, 1, 1, 0, 59, 0, 0, pytz.timezone(settings.TIME_ZONE)),data=[]):
        return {
            'template': {},
            'args': ({}, {}),
            'startTime': startTime,
            'endTime': endTime,
            'localOnly': False,
            'data': data
        }

    def _gen_series_list_with_data(self, key='collectd.test-db0.load.value', start=0, end=59, step=1, data=[]):
        seriesList = [
            TimeSeries(key, start, end, step, data)
        ]
        for series in seriesList:
            series.pathExpression = series.name

        return seriesList
```